examples/test-with-heavy-loads.rs

```diff
@@ -1,28 +1,29 @@
 use hyper::{Body, Response, Server};
 use routerify::{Middleware, Router, RouterService};
 use std::net::SocketAddr;
+use std::convert::Infallible;
 
-fn router() -> Router<Body, routerify::Error> {
+fn router() -> Router<Body, Infallible> {
     let mut builder = Router::builder();
 
     for i in 0..3000_usize {
         builder = builder.middleware(
             Middleware::pre_with_path(format!("/abc-{}", i), move |req| async move {
                 // println!("PreMiddleware: {}", format!("/abc-{}", i));
-                Ok(req)
+                Ok::<_, Infallible>(req)
```

Comment:
This is the extra type hinting you were mentioning?

Reply:
Yeah, it is necessary as otherwise Rust can't infer the return type of the closure